lib/gems/pending/util/log_dump_obj.rb

```diff
@@ -0,0 +1,35 @@
+module LogDumpObj
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def dumpObj(obj, prefix = nil, prnt_obj = STDOUT, prnt_meth = :puts, options = {})
+      meth = "dump#{obj.class.name}".to_sym
+      if self.respond_to?(meth)
+        prnt_obj.send(prnt_meth, "#{prefix}(#{obj.class}) = EMPTY") if obj.respond_to?(:blank?) && obj.blank?
+        send(meth, obj, prefix, prnt_obj, prnt_meth, options)
+      else
+        protected = false
+        if options[:protected].kind_of?(Hash)
+          protected = options[:protected][:path].to_miq_a.any? { |filter| prefix =~ filter }
+        end
+        if protected == true
+          prnt_obj.send(prnt_meth, "#{prefix}(#{obj.class}) = <PROTECTED>")
+        else
+          prnt_obj.send(prnt_meth, "#{prefix}(#{obj.class}) = #{obj.inspect}")
+        end
+      end
+    end
+
+    def dumpHash(hd, prefix, prnt_obj, prnt_meth, options)
+      hd.each { |k, v| dumpObj(v, "#{prefix}[#{k.inspect}]", prnt_obj, prnt_meth, options) }
+    end
+
+    def dumpArray(ad, prefix, prnt_obj, prnt_meth, options)
```

Comment:
@chessbyte I think because this is a refactoring work (see also https://github.com/ManageIQ/manageiq/pull/14547), the method names are unmodified for backward compatibility purpose. 

Maybe we can rename the method to a snake_case and make an alias for it.